app/uk/gov/hmrc/helptosavestub/controllers/BankDetailsBehaviour.scala

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.helptosavestub.controllers
+
+import play.api.libs.json.{JsValue, Json}
+import uk.gov.hmrc.helptosavestub.controllers.BARSController.{BARSResponse, BankDetails}
+import uk.gov.hmrc.helptosavestub.controllers.BankDetailsBehaviour.{CreateAccountResponse, Profile}
+import uk.gov.hmrc.helptosavestub.models.{ErrorDetails, NSIErrorResponse}
+
+trait BankDetailsBehaviour {
+
+  def getBankProfile(bankDetails: BankDetails): Profile = {
+    val accountNumberWithSortCodeIsValid =
+      if (bankDetails.accountNumber.startsWith("9")) {
+        Some(false)
+      } else if (bankDetails.accountNumber.startsWith("5")) {
+        None
+      } else {
+        Some(true)
+      }
+
+    val sortCodeIsPresentOnEISCD =
+      if (bankDetails.sortCode.startsWith("9")) {
+        "no"
+      } else if (bankDetails.sortCode.startsWith("5")) {
+        "whoopsie"
+      } else {
+        "yes"
+      }
+
+    if (bankDetails.accountNumber.startsWith("707")) {
+      Profile(None, Some(CreateAccountResponse(Left(NSIErrorResponse.incorrectAccountNumber))))
```

Comment:
here we'd want to do something like:
```scala
val barsResponse =
  accountNumberWithSortCodeIsValid.map(a => BARSResponse(a, sortCodeIsPresentOnEISCD))
```
and then stick `barsResponse` in all the `Profiles` below. That way, we can still get valid BARs responses when the create account response is negative.